src/main/java/com/google/sps/server/ScheduleCalendar.java

```diff
@@ -0,0 +1,250 @@
+
+   /**
+   * Creates a new secondary calendar to put a class schedule on.
+   * 
+   * @param client the user's Google calendar
+   */
+  public ScheduleCalendar(com.google.api.services.calendar.Calendar client) throws IOException {
+    try {
+      this.client = client;
+
+      // Get the timezone of the user's primary calendar
+      CalendarListEntry calendarListEntry = client.calendarList().get("primary").execute();
+      String calendarTimeZone = calendarListEntry.getTimeZone();
+
+      // Create a new calendar
+      com.google.api.services.calendar.model.Calendar newCalendar = new com.google.api.services.calendar.model.Calendar();
+      newCalendar.setSummary("class schedule");
+      newCalendar.setTimeZone(calendarTimeZone);
+
+      // Insert the new calendar
+      com.google.api.services.calendar.model.Calendar createdCalendar = client.calendars().insert(newCalendar).execute();
+      this.calendarId = createdCalendar.getId(); 
+    } catch (IOException e) {
+      System.out.println("Couldn't create secondary calendar");
+    }
+  }
+
+  /**
+   * Puts schedule onto secondary calendar
+   * 
+   * @param schedule the schedule to add to the calendar 
+   * @param calendarId the id of the secondary calendar to put the schedule on
+   */
+  public void addSchedule(Schedule schedule) throws IOException {
+    try {
+      // get current date, so can add the courses to the appropriate days of the current month 
+      java.util.Calendar calendar = java.util.Calendar.getInstance();
+      int nextSun = getDateOfNextSunday(calendar);
+      int month = getCurrentMonth(calendar);
+      int year = getCurrentYear(calendar);
+
+      // for each course in the schedule, add an event to the calendar
+      Iterator<Course> courseIterator = schedule.getCourses().iterator();
+      while (courseIterator.hasNext()) {
+        Course currentCourse = courseIterator.next();
+        List<TimeRange> sectionTimes = currentCourse.getSections().get(0).getMeetingTimes();
+        for (int i = 0; i < sectionTimes.size(); i++) {
+          String startTime = calculateStartTime(sectionTimes.get(i), month, year, nextSun);
+          String endTime = calculateEndTime(sectionTimes.get(i), month, year, nextSun);
+          addEvent(currentCourse, startTime, endTime);
+        }
+      } 
+    } catch (IOException e) {
+      System.out.println("Couldn't add schedule");
+    }   
+  }
+
+   /**
+   * To access the secondar calendar's id
+   *
+   * @return a String representing the id 
+   */
+  public String getCalendarId() {
+    return this.calendarId;
+  }
+
+   /**
+   * Helper function to calculate the day of the month of the next Sunday,
+   * to know when to start the schedule. 
+   *
+   * @param calendar an instance of Calendar, so can know the current date
+   * @return an int representing the date of next Sunday
+   */
+  public static int getDateOfNextSunday(java.util.Calendar calendar) {
```

Comment:
All of your static helper functions in this class can be moved to a separate CalendarUtils class since they're not dependent on having an instance of `ScheduleCalendar`. It's not 100% necessary, but since this file is getting a bit long it will help with code readability to have just the instance methods here: `getCalendarId`, `addEvent`, and `addSchedule`.